lib/templates/lab3/notifier.dart

```diff
@@ -19,16 +19,21 @@ class WaterIntakeNotifier extends StateNotifier<double> {
   }
 
   void _loadWaterIntake() async {
-    // TODO - Load the water intake from _localStorageService using await
-    // state =
-  }
+  state = await _localStorageService.getWaterIntake();
+}
 
   void increment(double amount) async {
-    state += amount;
-    // TODO - Save the water intake into _localStorageService using saveWaterIntake
-  }
+  if (state + amount <= 2000) {
```

Comment:
it is better to control that amout is less than max where you call increment, not here, when you use state